subsys/fs/littlefs_fs.c

```diff
@@ -727,6 +730,35 @@ static bool littlefs_compatible(const char *type_sz)
 	return (strcmp(type_sz, "LittleFS") == 0);
 }
 
+#ifdef CONFIG_FS_LITTLEFS_DEFAULT_FSDATA_ALLOCATOR
+
+/* Auto-mount dedicated section */
+struct fs_data_storage_t {
+	struct fs_littlefs lfs_data;
+	__aligned(4) uint8_t _read_buffer[CONFIG_FS_LITTLEFS_CACHE_SIZE];
+	__aligned(4) uint8_t _prog_buffer[CONFIG_FS_LITTLEFS_CACHE_SIZE];
+	uint32_t _lookahead_buffer[(CONFIG_FS_LITTLEFS_LOOKAHEAD_SIZE) /
+				   sizeof(uint32_t)];
+};
+
+static int fs_data_storage_init(struct fs_data_storage_t *res)
```

Comment:
This doesn't smell quite right.  We can only automount filesystems with default parameters?  I'd expect to see all the options exposed as devicetree properties.